jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java

```diff
@@ -509,15 +509,50 @@ protected void copyRequestHeaders(HttpServletRequest clientRequest, Request prox
 
     protected void addProxyHeaders(HttpServletRequest clientRequest, Request proxyRequest)
     {
-        addViaHeader(proxyRequest);
+        addViaHeader(clientRequest, proxyRequest);
         addXForwardedHeaders(clientRequest, proxyRequest);
     }
 
+    /**
+     * Adds the HTTP Via header to the proxied request.
+     *
+     * @deprecated Use {@link #addViaHeader(HttpServletRequest, Request)} instead.
+     * @param proxyRequest the request being proxied
+     */
+    @Deprecated
     protected void addViaHeader(Request proxyRequest)
     {
         proxyRequest.header(HttpHeader.VIA, "http/1.1 " + getViaHost());
     }
 
+    /**
+     * Adds the HTTP Via header to the proxied request, taking into account data present in the client request.
+     * This method considers the protocol of the client request when forming the proxied request. If it
+     * is HTTP, then the protocol name will not be included in the Via header that is sent by the proxy, and only
+     * the protocol version will be sent. If it is not, the entire protocol (name and version) will be included. 
+     * If the client request includes a Via header, the result will be appended to that to form a chain.
+     *
+     * @since 9.4.32
+     * @param clientRequest the client request
+     * @param proxyRequest the request being proxied
+     * @see <a href="https://tools.ietf.org/html/rfc7230#section-5.7.1">RFC 7230 section 5.7.1</a>
+     */
+    protected void addViaHeader(HttpServletRequest clientRequest, Request proxyRequest)
+    {
+        String protocol = clientRequest.getProtocol();
+        String[] parts = protocol.split("/", 2);
+        String protocolName = parts.length == 2 && "HTTP".equals(parts[0]) ? parts[1] : protocol;
+        String viaHeaderValue = "";
+        String clientViaHeader = clientRequest.getHeader(HttpHeader.VIA.name());
+
+        if (clientViaHeader != null)
+            viaHeaderValue = clientViaHeader;
+
+        viaHeaderValue += protocolName + " " + getViaHost();
+
+        proxyRequest.header(HttpHeader.VIA, viaHeaderValue);
+    }
```

Comment:
This method must call the old deprecated one.

Solution1:
* get `Via` header value
* call deprecated `addViaHeader()` method (now empty)
* get again `Via` header value; if it's changed, do nothing, else your new logic.

Solution2:
* call deprecated `addViaHeader()` method (now empty)
* your new logic

Opinions?

Reply:
Why? This is called from `addProxyHeaders`. For 1.1 clients, it's the same header value with one small change -- the protocol name for 1.1 is now uppercase. That's another bug though. The [protocol is case sensitive](https://tools.ietf.org/html/rfc7230#section-2.6) and the [registered name in IANA](https://www.iana.org/assignments/http-upgrade-tokens/http-upgrade-tokens.xhtml) is exactly `HTTP` not `http`. This fixes clients for 2.0 and other protocols. I think this is good to go. If you still disagree, can you elaborate?

Reply:
Ah, I see what you mean. They overrode `addViaHeader` and it's not called anymore by `addProxyHeaders`. 

Hum...

Reply:
Updated. How's that?